uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-legend.component.js

```diff
@@ -80,7 +80,7 @@ define([
       }, function (newValue, oldValue) {
         if (newValue !== oldValue) {
           $rootScope.$emit('LeaveCalendar::updateFiltersByAbsenceType',
-            vm.absenceTypesToFilterBy);
+            _.compact(vm.absenceTypesToFilterBy));
```

Comment:
Because of the generic absence type. It has an ID equal to an empty string ''. This is necessary to match it against leave requests with empty types. Also selecting the badge works, but it doesn't filter anything.

Reply:
All right, as the suggestion we may move the Leave to the next row alongside with "Weekend" etc. But this must be confirmed. I will let you know after we have a call.